src/main/groovy/nextflow/validation/parameters/ParameterValidator.groovy

```diff
@@ -153,10 +153,13 @@ class ParameterValidator {
             if (NF_OPTIONS.contains(param)) {
                 errors << "You used a core Nextflow option with two hyphens: '--${param}'. Please resubmit with '-${param}'".toString()
             }
-            else if (config.failUnrecognisedParams && !config.ignoreParams.contains(param)) {
-                errors << "* --${param.replaceAll("/", ".")}: ${getValueFromJsonPointer("/"+param, paramsJSON)}".toString()
-            } else if (!config.ignoreParams.contains(param)) {
-                warnings << "* --${param.replaceAll("/", ".")}: ${getValueFromJsonPointer("/"+param, paramsJSON)}".toString()
+            else if (!config.ignoreParams.any { param.startsWith(it) } ) {
```

Comment:
I don't think this is correct. I think that this would ignore e.g. `--genomes_extra` as well as `genomes`. 

I think that you need to use a check that looks like:

```groovy
ignogreParamsToks = config.ignoreParams.map{ it.tokenize('.') }

...

else if (!ignoreParamsToks.any{
    paramToks = param.tokenize('/')
    tokLen = it.size -1
    it == paramToks[0..tokLen]
})
``` 

or something or other.

Reply:
Good catch! I haven't thought about that edge case

Reply:
I've pushed a simpler version of your suggestion, would you mind taking another look?